backend/src/main/service/CachingService.ts

```diff
@@ -0,0 +1,52 @@
+import redis from 'redis';
+import { CachingObject } from '../data/CachingObject';
+
+export class CachingService {
+  // todo: change to env later
+  private client = redis.createClient({
+    // host: 'redis-server',
+    host: 'localhost',
+    port: 6379,
+  });
+
+  async cacheLatLonValues(
+    placeId: string,
+    latitude: number,
+    longitude: number
+  ): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      const value: CachingObject = { lat: latitude, lon: longitude };
+      this.client.setex(
+        placeId,
+        2592000,
+        JSON.stringify(value),
+        (err: Error | null, data: string) => {
+          if (err) {
+            reject(err);
+          }
+          resolve(true);
+        }
+      );
+    });
+  }
+
+  async getLatLonValue(placeId: string): Promise<CachingObject> {
+    return new Promise((resolve, reject) => {
+      this.client.get(placeId, (err: Error | null, data: string | null) => {
+        if (err) {
+          reject(err);
+        }
+
+        if (data) {
+          resolve(JSON.parse(data));
+        } else {
+          reject('No lat/lon information for this placeID');
+        }
+      });
+    });
```

Comment:
You mentioned that we don't cover the case yet of if the cache does not contain the placeId. Is this the place where we would add that behaviour?

LIke if the cache does not have the placeId, we would issue a request to get the new lat lon values so that we can cache again? Maybe we can leave a TODO or create an issue for that?

Reply:
Yup! I was planning to create an issue for it but basically thinking of putting it in here and making a new request to the FE Google Places API service

Reply:
Ah yup sure I would recommend creating the issue and then linking the issue in a todo comment inside the code. For example,

```suggestion
    return new Promise((resolve, reject) => {
      this.client.get(placeId, (err: Error | null, data: string | null) => {
        if (err) {
          // todo: https://github.com/ubclaunchpad/foodies/issues/89 ... some more details
          reject(err);
        }

        if (data) {
          resolve(JSON.parse(data));
        } else {
          reject('No lat/lon information for this placeID');
        }
      });
    });
```